pkg/workload/workload.go

```diff
@@ -315,6 +315,11 @@ func SetQuotaReservation(w *kueue.Workload, admission *kueue.Admission) {
 		evictedCond.Status = metav1.ConditionFalse
 		evictedCond.LastTransitionTime = metav1.Now()
 	}
+
+	// sync Admitted
+	if inSync, newCond := IsAdmittedInSync(w); !inSync {
+		apimeta.SetStatusCondition(&w.Status.Conditions, *newCond)
```

Comment:
Every time you call `IsAdmittedInSync` you also call `SetStatusCondition`.

I would change the function to be `SyncAdmittedCondition` that does both operations, returning True if something changed.

Reply:
done